### ISSUE TITLE
Skipping through songs 

### ISSUE THREAD

**Issue by LucyBrook** [Festify/app#320]
When playing on Microsoft edge songs skip through by a few minutes when playing. If you stop the current song and skip to the new one it doesn’t skip but the problem persists if you allow it to run.

**Comment by leolabs** [comment]
@LucyBrook I'll try to reproduce this on my side soon. Does this bug also occur in other browsers or is it limited to Edge?

**Comment by LucyBrook** [comment]
I have tried it in chrome and it also happens but less often

**Comment by TheDom42** [comment]
Could this be linked to #289 somehow? I was using Festify during a NYE party and people loved the concept.
However, we experienced the bug described in this issue as well as the one in #289 quite often.

This resulted in songs either a) being skipped right when it should start playing or b) skipping into the middle (or even to the last 10 seconds) of a song which kind of defeated the idea of the app.
Not sure if that is on the side of Festify or the Spotify API.
However, we only used Chrome - so I cannot say if it would have been worse with Edge.

**Comment by leolabs** [comment]
This is very strange. I know we had this issue as well at some points but it's hard to debug what exactly is causing this since it doesn't happen consistently.

**Comment by TheDom42** [comment]
Yes! That's true: very inconsistent - e.g. after a few hours it got noticeably better only to return an hour later. 

If haven't checked the code in detail but your main "database" is relying on firebase, right?
Could this be related to maybe a "sync" or race condition situation between firebase, your app and Spotify API? E.g. Spotify thinking that a song is already playing etc.? I really don't know much about your code but to me it seemed like some of the databases were just out of sync with each other.

**Comment by TowlieeNetworks** [comment]
i started using this in the office so everyone in the office can add their own songs, but we are experiencing the same issue. it mostly starts after an hour of playing. and indeed sometimes the whole song will be skipped, sometimes it will skip to the end or it will start playing in the middle of the song but correct it self after +/- 3sec and will go back to the start of the song.

is there any update on this issue so far?

**Comment by XLixl4snSU** [comment]
I'm experiencing the same right now sadly. Tested different browsers, persists. Removed all songs, paused, skipped etc, nothing changes...

**Comment by danstis** [comment]
Yeah, I've had this issue fairly consistently over the last couple of years. 
Tried from many machines, desktop/laptop/android phones and it happens on all of them. 

Seems to get worse after some time playing and I have never found a fix for it. 

I have tested excluding the site from the tab sleep in Edge, but no change.

**Comment by danstis** [comment]
I was able to reproduce this and capture the console output, in case it helps nail this one down.

For this set of logs, the current playing track finished as it should do, then the next song skipped to almost the end, for about half a second, then it started from the beginning. 

By the looks, the machine got a 502 response from the Spotify API, then retried (which is when the song restarted from the beginning.)

Here is the console log: https://pastebin.com/n73Kb5cG
